k2/csrc/intersect_dense_pruned.cu

```diff
@@ -698,6 +747,17 @@ class MultiGraphDenseIntersectPruned {
           K2_DCHECK_LT(static_cast<uint32_t>(scores_idx2),
                        static_cast<uint32_t>(scores_num_cols));
           float acoustic_score = scores_acc(scores_idx01, scores_idx2);
+          auto dest_state = arc.dest_state;
+          auto final_t = b_fsas_row_splits1[ai_fsa_idx0+1] - b_fsas_row_splits1[ai_fsa_idx0];
+          if (final_t - 1 == t && !has_valid_final_arc_data[ai_fsa_idx0] &&
+              allow_partial_) {
+              int32_t a_fsas_idx0 = a_fsas_row_ids1[sinfo.a_fsas_state_idx01];
+              // state_idx1 is 0-based.
+              // So "-1" is used when calculating a_fsas_final_state_idx1.
+              int32_t a_fsas_final_state_idx1 = a_fsas_row_splits1[a_fsas_idx0 + 1] - 1 - a_fsas_row_splits1[a_fsas_idx0];
+              dest_state = a_fsas_final_state_idx1;
+              acoustic_score = 0.0;
+          }
```

Comment:
I think this block and the above block are not necessary,  we can know which sequence has no final state by the shape of  ArcInfo at the last frame (see first K2_EVAL in FormatOutput),  at the last frame, there will be only one state (the final state) or no state at all. 
Another thing, if we modify `ai.u.dest_a_fsas_state_idx01` here, it will mess the arc-info and might raise an error for chunk by chunk decoding.  Actually, if we know which sequcence has no arc, we can set the dest-state of the arcs at the last frame to the extra state we added (see first K2_EVAL in FormatOutput) without knowing `ai.u.dest_info_state_idx1`.

Reply:
The last frame log_probs is manually added [0.0, -inf, -inf, -inf, ..., -inf, -inf].
The main purpose of this block is setting those -inf to 0.0.
Or all active arcs will be pruned by function `PruneTimeRange `

This is also the reason when the input num_frames == 20, while the generated lattice length is only 10!
Not only the final arc is missing, but also the last "10" frames are pruned by `PruneTimeRange `.

Before fix, lattice length is 10.
```
0 1 0 0 -8.41582e-05                                                                                                                                                        
1 2 0 0 -4.69674e-05                                                                                                                                                        
2 3 0 0 -9.17907e-06                                                                                                                                                        
3 4 0 0 -1.10864e-05                                                                                                                                                        
4 5 0 0 -1.4305e-05                                                                                                                                                         
5 6 0 0 -2.63449e-05                                                                                                                                                        
6 7 0 0 -6.55649e-06                                                                                                                                                        
7 8 0 0 -1.19209e-05                                                                                                                                                        
8 9 0 0 -3.12323e-05                                                                                                                                                        
9 10 0 0 -2.36032e-05                                                                                                                                                       
11
```

After fix the length is 20.
<img width="396" alt="image" src="https://user-images.githubusercontent.com/14951566/199647449-dce5dca3-4908-4c93-b6c3-21c28ccc284d.png">